src/features/hoverProvider.ts

```diff
@@ -21,7 +21,8 @@ export default class OmniSharpHoverProvider extends AbstractSupport implements H
 
         return serverUtils.typeLookup(this._server, req, token).then(value => {
             if (value && value.Type) {
-                let contents = [extractSummaryText(value.Documentation), { language: 'csharp', value: value.Type }];
+                let addLine = value.Documentation.split("\n").join("\n\n");
```

Comment:
I'm not convinced that this is correct. The bug wants us to have line breaks between sections (summary, parameter, etc), but this seems like it's going to add them anywhere a newline appears in the doc comment. Can you post a screenshot with your changes, with a doc comment with a multline summary section?

Reply:
![1057_6](https://user-images.githubusercontent.com/12554141/32916185-223dd55a-cad0-11e7-82c2-64a3e5b84d97.png)
Here is the output on adding a newline in the summary text. Isn't this what we are expecting?

Reply:
I don't think there should be a visible line break between "the" and "tag".

Reply:
![1057_6](https://user-images.githubusercontent.com/12554141/32916427-e57ecf42-cad0-11e7-8806-5bdbd01ce881.png)

An interesting observation here is that even if there are multiple newlines together in the text, vscode displays it as a single newline only. But I am not really sure whether this is ok or we need to do something about it.

Reply:
There are multiple "kinds" of newlines at play. 
* There are newlines that have to be in doc comments if a section spans more than one line. I contend that we should not cause a visible line break for those
* There might be multiple consecutive newlines in a doc comment section where the user was trying to separate paragraphs. We should make those visible.
* There are visible line breaks between sections (like between summary and parameters, or between parameters). We should always show those, even if the user wrote all their parameter documentation sections on one line.

Reply:
![1057_8](https://user-images.githubusercontent.com/12554141/32917539-543d955a-cad4-11e7-8bdc-3b5c498f7bec.png)

This is the output from the master branch(where no changes have been made). As noticeable, the  problem is that vscode doesn't show a newline for a single newline.It shows a "single" newline for 2 or more newlines(irrespective of whether they were part of the text originally or added by roslyn to separate the tags). Hence a quick solution was to double the '\n' character before being passed to the Hover object.

If we have to deal with single newlines and double newlines already present in the text separately, we will have to make changes on the roslyn side.Is that what you are suggesting?

Reply:
Yes, it looks like we will need to make a change that spans both repos. I propose that we:
* Augment Roslyn to be able to return some data structure that contains all of the "sections"
* Consume that on the VSCode side and properly insert newlines between sections.

Reply:
@rchande can we split this up into two features? @akshita31's change here makes the text much more readable than before, though it does not replicate VS's behavior. I'd start by finishing off this PR, then making it better once we have a chance to close on and implement the relevant change in omnisharp-roslyn. Thoughts?

Reply:
@TheRealPiotrP I pulled @akshita31's branch to play with the behavior. Given the doc comment in the first screenshot, this is what it looks like before and after the change, respectively. I feel that the second example is enough of a regression to the readability of longer documentation comments that we should wait until we have an appropriate change on the roslyn side. @DustinCampbell what do you think?

![image](https://user-images.githubusercontent.com/3751401/32957706-5dbbbaba-cb71-11e7-8e26-f41333570094.png)
![image](https://user-images.githubusercontent.com/3751401/32957711-6004f0ca-cb71-11e7-8e3f-6422df4a6327.png)

Reply:
I agree. Let's not merge changes that regress the user experience.